### PR TITLE
CNS-446: fix testutils block advancement simulation

### PR DESCRIPTION
**Pull request by oren-lava**
I made the NewBlock() function iterate over all keepers and call their corresponding BeginBlock() callback (instead of calling the inner functions directly). Also fixed some unit tests issues

### CODE REVIEW

File: testutil/keeper/keepers_init.go [hunk 1]
```diff
@@ -45,16 +45,17 @@ const (
 
 const BLOCK_HEADER_LEN = 32
 
+// Note: the keeper fields' order is consistent with the app's BeginBlockers order (search in app.go the phrase "app.mm.SetOrderBeginBlockers(")
```
Comment (orenl-lava) [inline testutil/keeper/keepers_init.go:48]: nit: replace with:
```
NOTE: the order of the keeper fields must follow that of calling app.mm.SetOrderBeginBlockers() in app/app.go
```

File: testutil/keeper/keepers_init.go [hunk 2]
```diff
@@ -278,10 +284,17 @@ func AdvanceEpoch(ctx context.Context, ks *Keepers, customBlockTime ...time.Dura
...
-	ks.Conflict.CheckAndHandleAllVotes(unwrapedCtx)
+	// get the value and type of the Keepers struct
+	keepersType := reflect.TypeOf(*ks)
+	keepersValue := reflect.ValueOf(*ks)
+
+	// iterate over all keepers and call BeginBlock (if it's implemented by the keeper)
+	for i := 0; i < keepersType.NumField(); i++ {
+		fieldValue := keepersValue.Field(i)
+
+		if BeginBlocker, ok := fieldValue.Interface().(KeeperBeginBlocker); ok {
```
Comment (orenl-lava) [inline testutil/keeper/keepers_init.go:296]: nit: s/BeginBlocker/beginBlocker/

File: x/conflict/keeper/keeper.go [hunk 3]
```diff
@@ -60,3 +60,7 @@ func (k Keeper) IsEpochStart(ctx sdk.Context) (res bool) {
 func (k Keeper) PushFixations(ctx sdk.Context) {
 	k.epochstorageKeeper.PushFixatedParams(ctx, 0, 0)
 }
+
+func (k Keeper) BeginBlock(ctx sdk.Context) {
```
Comment (orenl-lava) [inline x/conflict/keeper/keeper.go:64]: Module `conflict` does not appear in the BeginBlock registration in app.go. Should it?

File: x/epochstorage/keeper/keeper.go [hunk 4]
```diff
@@ -71,3 +72,16 @@ func (k *Keeper) AddFixationRegistry(fixationKey string, getParamFunction func(s
...
+
+func (k Keeper) BeginBlock(ctx sdk.Context) {
+	if k.IsEpochStart(ctx) {
+		// run functions that are supposed to run in epoch start
+		k.EpochStart(ctx)
+
+		// Notify world we have a new session
+
+		details := map[string]string{"height": fmt.Sprintf("%d", ctx.BlockHeight()), "description": "New Block Epoch Started"}
+		logger := k.Logger(ctx)
```
Comment (orenl-lava) [inline x/epochstorage/keeper/keeper.go:84]: you can embed "logger" directly in the next line.

File: x/pairing/keeper/keeper.go [hunk 5]
```diff
@@ -61,3 +61,10 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) IsEpochStart(ctx sdk.Context) (res bool) {
```
Comment (orenl-lava) [inline x/pairing/keeper/keeper.go:61]: drop this, and have caller use epochstorage's IsEpochStart()

File: x/spec/keeper/keeper.go [hunk 6]
```diff
@@ -42,3 +42,5 @@ func NewKeeper(
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+func (k Keeper) BeginBlock(ctx sdk.Context) {}
```
Comment (orenl-lava) [inline x/spec/keeper/keeper.go:46]: why is this needed? (the function does nothing)

File: x/subscription/keeper/keeper.go [hunk 7]
```diff
@@ -64,3 +64,10 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) IsEpochStart(ctx sdk.Context) bool {
```
Comment (orenl-lava) [inline x/subscription/keeper/keeper.go:64]: drop this; have caller use epochstorage's IsEpochStart()

**Review by orenl-lava — DISMISSED** [review]

File: testutil/keeper/keepers_init.go (hunk 2 above)
Comment (oren-lava) [reply testutil/keeper/keepers_init.go:296]: done

File: x/conflict/keeper/keeper.go (hunk 3 above)
Comment (oren-lava) [reply x/conflict/keeper/keeper.go:64]: not sure. @Yaroms ?

File: x/epochstorage/keeper/keeper.go (hunk 4 above)
Comment (oren-lava) [reply x/epochstorage/keeper/keeper.go:84]: done

File: x/pairing/keeper/keeper.go (hunk 5 above)
Comment (oren-lava) [reply x/pairing/keeper/keeper.go:61]: done

File: x/subscription/keeper/keeper.go (hunk 7 above)
Comment (oren-lava) [reply x/subscription/keeper/keeper.go:64]: done

File: x/spec/keeper/keeper.go (hunk 6 above)
Comment (oren-lava) [reply x/spec/keeper/keeper.go:46]: preparation for air-conditioning

File: x/conflict/keeper/keeper.go (hunk 3 above)
Comment (Yaroms) [reply x/conflict/keeper/keeper.go:64]: it is in the list

**Review by Yaroms — APPROVED** [review]
